docs/libvirt-pod-networking.md

```diff
@@ -4,7 +4,7 @@ Author: Fabian Deutsch \<fabiand@redhat.com\>
 
 ## Introduction
```

Comment:
Regarding the first paragraph: You are right that moving the IP into the VM is outside of the scope of what Kubernetes sees, but on the other hand Kubernetes also does not care. It just sees that IP bound to a Pod. What happens inside the pod to the interface or the IP, or how it got there is not something Kubernetes cares about. In the case of this proposal Kubernetes does not even see the IP to Pod assignment. CNI is pretty much Kubernetes unrelated. Kubernetes just uses it too.

Regarding the second paragraph (for migratable IPs): You are right that this proposal can't solve it, and I would not expect it. So a bad choosing of my example in that regard. This is really just a very limited approach to provide initial support, as far as I understood it (@fabiand?) . I for myself was more in favour of something like done in #160. There explicit support for calico was the goal. But for non-migratable VMs, the proposal here can be a good first step.

For migratable IPs, the only chance I see is explicit support added by us for the specific Kubernetes CNI provider (by directly talking with the network provider, without CNI, or by extending CNI).
If we want to use the k8s wirking objects and let the overlay network do the routing for us, we have to be on the same net like all pods. This also means that for some overlay network providers, we can't have this direct support (e.g. flannel, which only works with preassigned pools). For others like calico an ovn, we can.